### ISSUE TITLE
can you provide the training log?

### ISSUE THREAD

**Issue by argman** [zhaoweicai/mscnn#33]
Thanks for the nice work, as i have made some changes to the code, i'd like to check whether there are bugs in my code, thanks!

**Comment by argman** [comment]
during the training, the loss is a bit big as below
```
I1207 11:31:41.510645  1124 solver.cpp:231] Iteration 14450, loss = 11333
I1207 11:31:41.510880  1124 solver.cpp:247]     Train net output #0: accuracy_1_5x5 = 0
I1207 11:31:41.510898  1124 solver.cpp:247]     Train net output #1: accuracy_1_5x5 = 0
I1207 11:31:41.510907  1124 solver.cpp:247]     Train net output #2: accuracy_1_7x7 = 0
I1207 11:31:41.510920  1124 solver.cpp:247]     Train net output #3: accuracy_1_7x7 = -1
I1207 11:31:41.510926  1124 solver.cpp:247]     Train net output #4: accuracy_2_5x5 = 0
I1207 11:31:41.510934  1124 solver.cpp:247]     Train net output #5: accuracy_2_5x5 = 0
I1207 11:31:41.510942  1124 solver.cpp:247]     Train net output #6: accuracy_2_7x7 = 0
I1207 11:31:41.510949  1124 solver.cpp:247]     Train net output #7: accuracy_2_7x7 = 0
I1207 11:31:41.510957  1124 solver.cpp:247]     Train net output #8: accuracy_3_5x5 = 0
I1207 11:31:41.510964  1124 solver.cpp:247]     Train net output #9: accuracy_3_5x5 = -1
I1207 11:31:41.510972  1124 solver.cpp:247]     Train net output #10: accuracy_3_7x7 = 0
I1207 11:31:41.510979  1124 solver.cpp:247]     Train net output #11: accuracy_3_7x7 = 0
I1207 11:31:41.510987  1124 solver.cpp:247]     Train net output #12: accuracy_4_5x5 = 0
I1207 11:31:41.510994  1124 solver.cpp:247]     Train net output #13: accuracy_4_5x5 = -1
I1207 11:31:41.511023  1124 solver.cpp:247]     Train net output #14: boxiou_1_5x5 = 0.0485884
I1207 11:31:41.511031  1124 solver.cpp:247]     Train net output #15: boxiou_1_7x7 = -1
I1207 11:31:41.511040  1124 solver.cpp:247]     Train net output #16: boxiou_2_5x5 = 0.0703838
I1207 11:31:41.511049  1124 solver.cpp:247]     Train net output #17: boxiou_2_7x7 = 0.0371105
I1207 11:31:41.511056  1124 solver.cpp:247]     Train net output #18: boxiou_3_5x5 = -1
I1207 11:31:41.511065  1124 solver.cpp:247]     Train net output #19: boxiou_3_7x7 = 0.038233
I1207 11:31:41.511072  1124 solver.cpp:247]     Train net output #20: boxiou_4_5x5 = -1
I1207 11:31:41.511096  1124 solver.cpp:247]     Train net output #21: loss_1_5x5 = 2554.64 (* 0.9 = 2299.17 loss)
I1207 11:31:41.511119  1124 solver.cpp:247]     Train net output #22: loss_1_5x5 = 0.0436281 (* 0.9 = 0.0392653 loss)
I1207 11:31:41.511129  1124 solver.cpp:247]     Train net output #23: loss_1_7x7 = 931.59 (* 0.9 = 838.431 loss)
I1207 11:31:41.511137  1124 solver.cpp:247]     Train net output #24: loss_1_7x7 = 0 (* 0.9 = 0 loss)
I1207 11:31:41.511147  1124 solver.cpp:247]     Train net output #25: loss_2_5x5 = 1965.12 (* 1 = 1965.12 loss)
I1207 11:31:41.511155  1124 solver.cpp:247]     Train net output #26: loss_2_5x5 = 0.0477147 (* 1 = 0.0477147 loss)
I1207 11:31:41.511164  1124 solver.cpp:247]     Train net output #27: loss_2_7x7 = 2379.96 (* 1 = 2379.96 loss)
I1207 11:31:41.511204  1124 solver.cpp:247]     Train net output #28: loss_2_7x7 = 0.0422432 (* 1 = 0.0422432 loss)
I1207 11:31:41.511214  1124 solver.cpp:247]     Train net output #29: loss_3_5x5 = 858.809 (* 1 = 858.809 loss)
I1207 11:31:41.511224  1124 solver.cpp:247]     Train net output #30: loss_3_5x5 = 0 (* 1 = 0 loss)
I1207 11:31:41.511234  1124 solver.cpp:247]     Train net output #31: loss_3_7x7 = 2248.98 (* 1 = 2248.98 loss)
I1207 11:31:41.511242  1124 solver.cpp:247]     Train net output #32: loss_3_7x7 = 0.0609456 (* 1 = 0.0609456 loss)
I1207 11:31:41.511252  1124 solver.cpp:247]     Train net output #33: loss_4_5x5 = 742.361 (* 1 = 742.361 loss)
I1207 11:31:41.511260  1124 solver.cpp:247]     Train net output #34: loss_4_5x5 = 0 (* 1 = 0 loss)
```
Is this normal ?

**Comment by argman** [comment]
I find the preprocess in image_gt_data.cpp is complex, can you make a brief explain about the steps, thanks!

**Comment by zhaoweicai** [comment]
You can check my log files of `mscnn-7s-576-2x` for KITTI car: [1st stage](http://www.svcl.ucsd.edu/projects/mscnn/log_1st.txt) and [2nd stage](http://www.svcl.ucsd.edu/projects/mscnn/log_2nd.txt).

Before you make some changes, I strongly suggest you run the demo code. You can have these log files by simply running the demo code. 

In `image_gt_data_layer.cpp`, it has some comments, which is sufficient for you to know what each part is doing. I may not have time for detailed comments.

**Comment by argman** [comment]
@zhaoweicai ,thanks for your reply, I will try as you suggested.
